Assets/MixedRealityToolkit.Examples/Demos/Solvers/Scripts/SolverExampleManager.cs

```diff
@@ -69,24 +69,29 @@ public void SetOrbital()
 
         public void SetSurfaceMagnetism()
         {
-            DestroySolver();
+            DisableSolver();
 
-            AddSolver<SurfaceMagnetism>();
+            EnableSolver<SurfaceMagnetism>();
 
             // Modify properties of solver custom to this example
             var surfaceMagnetism = currentSolver as SurfaceMagnetism;
             surfaceMagnetism.SurfaceNormalOffset = 0.2f;
         }
 
-        private void AddSolver<T>() where T : Solver
+        private void EnableSolver<T>() where T : Solver
         {
-            currentSolver = gameObject.AddComponent<T>();
-            handler = GetComponent<SolverHandler>();
+            currentSolver = gameObject.EnsureComponent<T>();
+            currentSolver.enabled = true;
             RefreshSolverHandler();
         }
 
         private void RefreshSolverHandler()
         {
+            if(handler == null)
```

Comment:
What if the handler has changed programatically?  This would cause the hander to NOT be refreshed

Reply:
I can't follow, the following code also just takes the assumption that `handler` is still referencing the correct handler, and does not gather the correct handler reference every time.
This is for (a) it is the first call and the SolverHandler component has just been added to the GameObject, (b) someone/something has removed the SolverHandler component, causing a new component to be added.

If the handler has been changed programatically, the `handler` reference wouldn't be `null` therefore this is skipped.